pkg/server/api/svid/v1/server_test.go

```diff
@@ -2,35 +2,88 @@ package svid_test
 
 import (
 	"context"
+	"crypto/rand"
+	"crypto/x509"
+	"net/url"
 	"testing"
+	"time"
 
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/server/api"
+	"github.com/spiffe/spire/pkg/server/api/rpccontext"
 	"github.com/spiffe/spire/pkg/server/api/svid/v1"
 	svidpb "github.com/spiffe/spire/proto/spire-next/api/server/svid/v1"
 	"github.com/spiffe/spire/test/spiretest"
+	"github.com/spiffe/spire/test/testkey"
 	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
 )
 
-func TestServer(t *testing.T) {
-	registerFn := func(s *grpc.Server) {
-		svid.RegisterService(s, FakeService{})
-	}
+func TestMintX509SVID(t *testing.T) {
+	log, _ := test.NewNullLogger()
+	ctx := rpccontext.WithLogger(context.Background(), log)
```

Comment:
I think it's confusing to create this context here, embellish it with things intended for the server, and then use it to make the RPC call with the client. It's also problematic that the request context is being dropped in `contextFn`, we should extend that context and return it instead of dropping it and returning a new one.

I'd suggest that:
- The `createClient` function not take a context, but rather actively embellish the request context via the `contextFn` callback.
- The null logger be created inside of `createClient` unless we want to test that the right stuff is logged, in which case, `createClient` could return the `hook` returned from `test.NewNullLogger`